Add op2ext as a submodule and hook to Visual Studio project files

Brett208 opened the pull request:
Had to bring in OP2Internal as op2ext's submodule and hook to NetFixClient's project files. Otherwise, op2ext would not properly compile.

Will look at the recommendation about the environment variable later or in a different branch.

Closes #24 

-Brett

DanRStevens commented:
AppVeyor builds are failing.

My initial thought was it wasn't cloning the new submodule, but it actually is. This project has the AppVeyor config update to clone submodules already, as it was needed for OP2Internal.

My next guess is the project references are not set correctly.

I didn't realize op2ext depends on Outpost2DLL. I'd forgotten about that changes. This brings in more dependencies than I expected. Not quite sure what your comment was saying about OP2Internal. Did you mean Outpost2DLL?

Brett208 commented:
My vague comment was trying to convey that I was getting local build errors until I added op2ext as a project to the NetFixClient solution file. Then it was able to build properly.

I just cloned a fresh copy of the NetFixClient repository at a different location, switched to this branch, recursively updated the submodules, and then compiled. Worked perfectly.

I don't know what the issue is, but the project settings translate fine locally. Must be a difference between me building it and appveyor.

It says 'Solution project file "op2ext.vcxproj" cannot be found.' Is there a way to navigate appveyor's downloaded files and make sure that it pulled that file?

DanRStevens commented on NetFixClient.sln:
```diff
@@ -7,6 +7,10 @@ Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "NetFixClient", "NetFixClien
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "OP2Internal", "OP2Internal\OP2Internal.vcxproj", "{97594BB1-793F-379B-F6AB-040AC8FE1424}"
 EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "op2ext", "..\op2ext\op2ext.vcxproj", "{6DE3610C-C9EE-4FD3-BBD7-66382C16FDE9}"
```
This path doesn't look right. Why the "..\\" here? That would place the op2ext project above the level of the solution file, rather than below it.

I suspect you are referencing a different copy of op2ext that exists locally on your system, which is outside of the NetFixClient folder, which is why it compiles for you, but fails on the CI servers.

DanRStevens commented:
From the AppVeyor logs, the cloned repository paths can be seen:
```
git submodule update --init --recursive
Submodule 'OP2Internal' (https://github.com/OutpostUniverse/OP2Internal.git) registered for path 'OP2Internal'
Submodule 'op2ext' (https://github.com/OutpostUniverse/op2ext.git) registered for path 'op2ext'
Cloning into 'C:/projects/netfixclient/OP2Internal'...
Cloning into 'C:/projects/netfixclient/op2ext'...
Submodule path 'OP2Internal': checked out '17b6ea5cb0e5a0f32e4832547836f82948dd34eb'
Submodule path 'op2ext': checked out 'de63d9d0c37e5af82b9093ccd5be4bc42a5a873d'
Submodule 'Submodules/Outpost2DLL' (https://github.com/OutpostUniverse/Outpost2DLL.git) registered for path 'op2ext/Submodules/Outpost2DLL'
Cloning into 'C:/projects/netfixclient/op2ext/Submodules/Outpost2DLL'...
Submodule path 'op2ext/Submodules/Outpost2DLL': checked out '0667787bddb8ac68bb5a09b21b6cecb18cd44d0a'
```

The error seems to be the `..\` in the path reference to op2ext. Your computer is referencing a copy of op2ext installed in the same parent folder as NetFixClient, rather than as a subfolder of NetFixClient.

That's also why you can do a fresh clone and have it still work. Your fresh clone is probably in the same parent folder.

----

I assume the local build errors you were getting were due to changes to the logging code. There should be no current dependency between the projects.

I suppose I'm stating that so we remain clear on why we're adding this new submodule dependency.

Brett208 commented:
Well, I feel a little dumb for not figuring that out on my own. Thanks for pointing it out.
